packages/ra-ui-materialui/src/layout/PageTitleConfigurable.tsx

```diff
@@ -34,7 +39,24 @@ export const PageTitleConfigurable = ({ preferenceKey, ...props }) => {
                 },
             }}
         >
-            <PageTitle {...props} />
+            <PageTitleConfigurableInner {...props} />
         </Configurable>
     );
 };
+
+const PageTitleConfigurableInner = props => {
```

Comment:
The reason is actually more to do with the `<PageTitle />` component.

Previously, the `<PageTitle />` component always checks preferences and is only used in `<PageTitleConfigurable />`.

Now, `<PageTitle />` also needs to be used in a `nonConfigurable` context, so it cannot check preferences, which is why I removed that logic from `<PageTitle />` (I thought `PageTitle` was a good name for a non configurable title, so I repurposed it instead of creating a new component).

Now, the logic to check preferences (`const [titleFromPreferences] = usePreference();`) needs to live somewhere. It can't live directly in `PageTitleConfigurable` since I think it needs to live in a child of `Configurable`. Thus, I needed a child component that checks preferences and renders `<PageTitle />` and that's what `<PageTitleConfigurableInner />` is for.

Please let me know if that's clear or if there's a better way to do this.

Reply:
Super clear, thanks for the explanation.